### PR TITLE
Add Json.traverse* methods

### PR DESCRIPTION
**Pull request by sergeykolbasov**
It's similar to `Json.map*` transformations but with a drop of `F[_] : Applicative`

We mangle a lot with JSON AST, often in some monadic context. This collection of methods would come in handy.

### CODE REVIEW

**Comment by zarthross** [comment]
Thanks for the PR.  Love the idea.  What do you think of changing the signature though to be a little more flexible.

From:
`def traverseBoolean[F[_]](f: Boolean => F[Boolean])(implicit F: Applicative[F]): F[Json]`

To:
`def traverseBoolean[F[_]](f: Boolean => F[Json])(implicit F: Applicative[F]): F[Json]`

I think this would allow more transformation within the monadic context.  This would make it more akin to the `with*` methods than the `map*` methods, but you get a greater flexibility.

**Comment by hamnis** [comment]
I think this would a nice set of combinators to have, @sergeykolbasov are you willing to make the changes suggested by @zarthross ?

The boolean case is an example, it would be the same for all the new traverseX functions

**Comment by hamnis** [comment]
mimaErrors are genuine. We do not have these in previous releases, so we need to add exclusions rules here.

**Review by hamnis — DISMISSED** [review]

File: modules/core/shared/src/main/scala/io/circe/Json.scala [hunk 1]
```diff
@@ -350,6 +365,14 @@ object Json {
     final def mapString(f: String => String): Json = this
     final def mapArray(f: Vector[Json] => Vector[Json]): Json = this
     final def mapObject(f: JsonObject => JsonObject): Json = this
+
+    final def traverseBoolean[F[_]](f: Boolean => F[Json])(implicit F: Applicative[F]): F[Json] =
+      F.map(f(value))(identity)
```
Comment (zarthross) [inline modules/core/shared/src/main/scala/io/circe/Json.scala]: I think we might be overly complicating the traverse methods here that are not just pure.  Here is my suggestion:

```suggestion
    final def traverseBoolean[F[_]](f: Boolean => F[Json])(implicit F: Applicative[F]): F[Json] = f(value)
```

This change would need to be applied to each of there other sum types here.

Comment (zarthross) [reply modules/core/shared/src/main/scala/io/circe/Json.scala]: Pushed a fix for this.

**Review by zarthross — APPROVED** [review]

**Pull request merged**